### PR TITLE
Support zeros_like() for NT

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96354
* #96528
* __->__ #96527
* #96211

This is used for the fake tensor fallbacks.

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -490,15 +490,26 @@ def test_fill_(self):
...
-
-        for nt_ub in ones_nt.unbind():
-            t = torch.ones_like(nt_ub)
+        nt.zero_()
+        for nt_ub in nt.unbind():
+            t = torch.empty_like(nt_ub)
+            t.fill_(0.)
             self.assertEqual(nt_ub, t)
 
+    @parametrize("func", [torch.ones_like, torch.zeros_like],
```
Comment (cpuhrsch) [inline test/test_nestedtensor.py:502]: If this introduces spaces or weird symbols the internal test runner might not like it. You can change the display name by using explicit names. See https://github.com/pytorch/pytorch/blob/master/test/test_sparse_csr.py#L207

Comment (jbschlosser) [reply test/test_nestedtensor.py:502]: on the next line, I use `name_fn=lambda f: f.__name__` to ensure readable test names :)

**Pull request closed without merging**